### PR TITLE
feat(ONYX-1293): redesign save cta and add follow cta

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [ONYX-1293] [ONYX-1286] 

### Description

Redesign Save CTA and add Follow CTA to the artwork card and grid item

Implement experiments: variant-a, variant-b and variant-c
- *Control variation*: app users will see rail cards with current design with the save button in its original position.
- *Variant B:* app users will see rail cards with save button below.
- *Variant C:* app users will see rail cards with save and follow button below.

Add confirmation toast for the Artist follow action

Hide changes behind a feature flag AREnableNewSaveAndFollowOnArtworkCard

Figma: https://www.figma.com/design/tZ5OjG3dXFmNKFY3zqXlYV/App-Card-and-Rail?node-id=411-12304&node-type=canvas&m=dev
## Rail
| iOS variant-a | iOS variant-b | iOS variant-c |
| --- | --- | --- |
| image | image | image |

| Android variant-a | Android variant-b | Android variant-c |
| --- | --- | --- |
| Screenshot_1730992571 | Screenshot_1730992635 | Screenshot_1730992702 |

## Grid with 1 column
| iOS variant-a | iOS variant-b | iOS variant-c |
| --- | --- | --- |
| image | image | image |

| Android variant-a | Android variant-b | Android variant-c |
| --- | --- | --- |
| Screenshot_1730992546 | Screenshot_1730992640 | Screenshot_1730992696 |

## Grid with 2 columns
| iOS variant-a | iOS variant-b | iOS variant-c |
| --- | --- | --- |
| image | image | image |

| Android variant-a | Android variant-b | Android variant-c |
| --- | --- | --- |
| Screenshot_1730992550 | Screenshot_1730992638 | Screenshot_1730992699 |

## Follow Artist toast
| iOS | Android |
| --- | --- |
| image | Screenshot_1730992767 |

## Save button on Artwork screen
| Before Liked | Before not Liked | After (experiment b or c) Liked | After (experiment b or c) not Liked |
| --- | --- | --- | --- |
| image | image | image | image |

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- redesign Save CTA and add Follow CTA, use A/B test with 3 variants - daria

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1293]: https://artsyproduct.atlassian.net/browse/ONYX-1293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[ONYX-1286]: https://artsyproduct.atlassian.net/browse/ONYX-1286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dariakoko** [comment]
Note: the tracking of the experiment has to be adjusted: we need to track the experiment on the ArtworksRail component and track once per session. Will be addressed in a separate PR

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (redesign Save CTA and add Follow CTA, use A/B test with 3 variants - daria - dariakoko)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d8fdd80fb126d3cec0f4a1cb683b4afa72531cd8
</p>

File: src/app/Components/Artist/ArtistHeaderNavRight.tsx [hunk 1]
```diff
@@ -20,8 +21,11 @@ export const ArtistHeaderNavRight: React.FC<ArtistHeaderNavRightProps> = ({
   const data = useFragment(fragment, artist)
   const [isFollowed, setIsFollowed] = useState(!!data?.isFollowed)
 
-  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-  const { handleFollowToggle } = useFollowArtist(data!)
+  const { handleFollowToggle } = useFollowArtist({
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    artist: data!,
```
Comment (olerichter00) [inline src/app/Components/Artist/ArtistHeaderNavRight.tsx:26]: I think we could instead update the artist parameter type for `useFollowArtist` to `artist: useFollowArtist_artist$key | null`.

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx [hunk 2]
```diff
@@ -117,13 +120,31 @@ export const Artwork: React.FC<ArtworkProps> = ({
...
+
+  const enableShowOldSaveCTA =
+    newSaveAndFollowOnArtworkCardExperiment.enabled &&
+    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-a"
+  const enableNewSaveCTA =
+    newSaveAndFollowOnArtworkCardExperiment.enabled &&
+    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-b"
+  const enableNewSaveAndFollowCTAs =
+    newSaveAndFollowOnArtworkCardExperiment.enabled &&
+    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-c"
```
Comment (olerichter00) [inline src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:147]: Would it work to simplify this code block like this and remove the `enableNewSaveAndFollowOnArtworkCard` feature flag checks further down in the code?

```suggestion
  const enableNewSaveAndFollowOnArtworkCard = useFeatureFlag(
    "AREnableNewSaveAndFollowOnArtworkCard"
  )
  const newSaveAndFollowOnArtworkCardExperiment = useExperimentVariant(
    "onyx_artwork-card-save-and-follow-cta-redesign"
  )

  const enableShowOldSaveCTA =
  !enableNewSaveAndFollowOnArtworkCard ||
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-a"
  const enableNewSaveCTA =
  enableNewSaveAndFollowOnArtworkCard &&
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-b"
  const enableNewSaveAndFollowCTAs =
  enableNewSaveAndFollowOnArtworkCard &&
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-c"
```

File: src/app/Scenes/HomeView/HomeView.tsx [hunk 3]
```diff
@@ -128,6 +129,13 @@ export const HomeView: React.FC = () => {
     })
   }
 
+  // TODO: track on a Rail, use useHomeViewExperimentTracking (adjust the hook beforehand)
```
Comment (olerichter00) [inline src/app/Scenes/HomeView/HomeView.tsx:132]: 👀

File: src/app/Scenes/HomeView/hooks/useHomeViewExperimentTracking.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import { compact } from "lodash"
 import { useEffect } from "react"
 import { useTracking } from "react-tracking"
 
+// TODO: track only once per session
```
Comment (olerichter00) [inline src/app/Scenes/HomeView/hooks/useHomeViewExperimentTracking.ts:8]: 👀

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx (hunk 2 above)
Comment (dariakoko) [reply src/app/Components/ArtworkGrids/ArtworkGridItem.tsx]: That would be better! I extracted this logic into a separate function

File: src/app/Components/Artist/useFollowArtist.ts [hunk 5]
```diff
@@ -89,10 +105,24 @@ export const useFollowArtist = (artist: useFollowArtist_artist$key) => {
...
+      !data.isFollowed &&
+      toast.show("Artist Followed", "bottom", {
+        cta: "View Follows",
+        onPress: () => {
+          switchTab("profile")
+          navigate("favorites")
+        },
+        backgroundColor: "green100",
+        description: "Keep track of the artists you love",
+      })
```
Comment (MounirDhahri) [inline src/app/Components/Artist/useFollowArtist.ts:125]: **nitpick:** I prefer old fashioned if statements here because I find them more readable
```
if (showToast && !data.isFollowed) {
  toast.show(...)
}
```

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx [hunk 6]
```diff
@@ -443,22 +461,40 @@ export const Artwork: React.FC<ArtworkProps> = ({
...
-                    testID="save-artwork-icon"
-                  >
-                    <ArtworkHeartIcon isSaved={!!isSaved} index={itemIndex} />
-                  </Touchable>
-                </Flex>
-              )}
+              {!hideSaveIcon &&
+                (!enableNewSaveAndFollowOnArtworkCard ||
+                  !newSaveAndFollowOnArtworkCardExperiment.enabled ||
+                  !!enableShowOldSaveCTA) && (
```
Comment (MounirDhahri) [inline src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:467]: **suggestion:** I suggest we move this to a separate constant to simplify understanding the render in this component. Maybe we can do something like 
```typescript
cont showX = hideSaveIcon && (!enableNewSaveAndFollowOnArtworkCard || !newSaveAndFollowOnArtworkCardExperiment.enabled || !!enableShowOldSaveCTA)
```
that still feels too complicated to me but it will make understanding the render method easier. 


we only recently simplified this component, I would suggest to keep in mind that we can get again quickly to where we were and end up having this component render complicated again.

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 7]
```diff
@@ -86,7 +83,7 @@ export const ArtworkRail: React.FC<ArtworkRailProps> = ({
             showSaveIcon={showSaveIcon}
             hideIncreasedInterestSignal={hideIncreasedInterestSignal}
             hideCuratorsPickSignal={hideCuratorsPickSignal}
-            {...extractArtworkActionTrackingProps(otherProps)}
```
Comment (MounirDhahri) [inline src/app/Components/ArtworkRail/ArtworkRail.tsx:89]: is this change intentional?

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx
Comment (MounirDhahri) [inline src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:1]: I see a few changes in this file but no corresponding updates for tests. Sometimes that's fine but in this case the `ArtworkGridItem` is a key component and would suggest adding test cases here to cover our bases and avoid future regressions

File: src/app/Components/ArtworkRail/ArtworkRailCard.tsx [hunk 8]
```diff
@@ -63,6 +80,13 @@ export const ArtworkRailCard: React.FC<ArtworkRailCardProps> = ({
     ;(artwork as DissapearableArtwork)?._disappearable?.disappear()
   }
 
+  // 36 = 20 (padding) + 16 (icon size) + 5 (top padding)
+  const likeAndFollowCTAPadding =
+    enableNewSaveAndFollowOnArtworkCard && (enableNewSaveCTA || enableNewSaveAndFollowCTAs) ? 41 : 0
+  const artworkRailCardMetaPadding = 10
+  const artworkRailCardMetaDataHeight =
+    ARTWORK_RAIL_TEXT_CONTAINER_HEIGHT + artworkRailCardMetaPadding + likeAndFollowCTAPadding // TODO: check height
```
Comment (MounirDhahri) [inline src/app/Components/ArtworkRail/ArtworkRailCard.tsx:88]: do you still need to check the height here 👀 
```suggestion
    ARTWORK_RAIL_TEXT_CONTAINER_HEIGHT + artworkRailCardMetaPadding + likeAndFollowCTAPadding // TODO: check height
```

Comment (MounirDhahri) [inline src/app/Components/ArtworkRail/ArtworkRailCard.tsx:88]: this is scary 😄  but thanks for explaining the math behind it

File: src/app/Components/ArtworkRail/ArtworkRailCardMeta.tsx [hunk 9]
```diff
@@ -61,6 +62,17 @@ export const ArtworkRailCardMeta: React.FC<ArtworkRailCardMetaProps> = ({
...
+    "AREnableNewSaveAndFollowOnArtworkCard"
+  )
+  const newSaveAndFollowOnArtworkCardExperiment = useExperimentVariant(
+    "onyx_artwork-card-save-and-follow-cta-redesign"
+  )
+
+  const { enableShowOldSaveCTA } = getExperimentVariant(
+    newSaveAndFollowOnArtworkCardExperiment.enabled,
+    newSaveAndFollowOnArtworkCardExperiment.variant
+  )
```
Comment (MounirDhahri) [inline src/app/Components/ArtworkRail/ArtworkRailCardMeta.tsx:75]: fwiw, I think you can even simplify this more by moving `useExperiementVariant` and `useFeatureFlag` to the hook
```suggestion
  const {enableShowOldSaveCTA} = useArtworkSomethingExperiment()
```

File: src/app/Scenes/Artwork/Components/ArtworkItemCTAs.tests.tsx
Comment (MounirDhahri) [inline src/app/Scenes/Artwork/Components/ArtworkItemCTAs.tests.tsx:1]: thank you for the tests

File: src/app/store/migration.ts [hunk 10]
```diff
@@ -57,9 +57,10 @@ export const Versions = {
   DeleteDirtyArtworkDetails: 44,
   AddSubmissionDraft: 45,
   DeleteArtworkAndArtistViewOption: 46,
+  AddTrackedSectionTypes: 47,
 }
 
-export const CURRENT_APP_VERSION = Versions.DeleteArtworkAndArtistViewOption
+export const CURRENT_APP_VERSION = Versions.AddTrackedSectionTypes
```
Comment (MounirDhahri) [inline src/app/store/migration.ts:63]: **issue:** this is not needed since we don't keep the `HomeViewStoreProvider` after closing the app. We only need to write migrations for values in global store

File: src/app/system/devTools/DevMenu/Components/ExperimentFlagItem.tsx [hunk 11]
```diff
@@ -105,23 +105,45 @@ export const ExperimentFlagItem: React.FC<{ description: string; flag: EXPERIMEN
...
-                      </Pill>
-                      <Pill
-                        variant="badge"
-                        onPress={() => {
-                          setVariant("experiment")
-                        }}
-                      >
-                        experiment {experiment.unleashVariant === "experiment" && "(default)"}
-                      </Pill>
+
```
Comment (MounirDhahri) [inline src/app/system/devTools/DevMenu/Components/ExperimentFlagItem.tsx:108]: thanks for this addition

File: src/app/utils/experiments/getExperimentVariant.tsx [hunk 12]
```diff
@@ -0,0 +1,26 @@
+export const getExperimentVariant = (enabled: boolean, variant: string) => {
```
Comment (MounirDhahri) [inline src/app/utils/experiments/getExperimentVariant.tsx:1]: **issue:** is this really needed? I would rather suggest we rely on `useExperimentVariant` instead to avoid maintaining two methods that more or less do the same thing. It also works pretty well with our overrides unlike this new method which will also need to be updated to support it.

If there is something broken with `useExperimentVariant`, then we have a big problem and we need to fix it 😄 .

Ideally we would to get the variant like this or so:
```
  const experiment = useExperimentVariant(flag)
  const variant = experiment.variant
```

File: src/app/utils/experiments/hooks.ts [hunk 13]
```diff
@@ -43,22 +43,22 @@ export function useExperimentVariant(name: EXPERIMENT_NAME): {
...
     if (!enabled) {
       return
     }
 
     reportExperimentVariant({
       experimentName: name,
-      variantName: variant?.name ?? "default-variant",
-      payload: variant?.payload?.value,
+      variantName: localVariantOverrides[name] || unleashVariant,
+      payload: localPayloadOverrides[name] || unleashPayload,
```
Comment (MounirDhahri) [inline src/app/utils/experiments/hooks.ts:57]: nice

File: src/app/utils/experiments/getExperimentVariant.tsx (hunk 12 above)
Comment (dariakoko) [reply src/app/utils/experiments/getExperimentVariant.tsx:1]: I added it to avoid implementing the same variant check in multiple files:
```
  const newSaveAndFollowOnArtworkCardExperiment = useExperimentVariant(
    "onyx_artwork-card-save-and-follow-cta-redesign"
  )
  const enableShowOldSaveCTA =
    newSaveAndFollowOnArtworkCardExperiment.enabled &&
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-a"
  const enableNewSaveCTA =
    newSaveAndFollowOnArtworkCardExperiment.enabled &&
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-b"
  const enableNewSaveAndFollowCTAs =
    newSaveAndFollowOnArtworkCardExperiment.enabled &&
    newSaveAndFollowOnArtworkCardExperiment.payload === "variant-c" 
```

It will only be used for this experiment and will be removed when the experiment is finished.
The experiment is taking a long time and I'm afraid, the logic can be messed up eventually is we keep these code duplicated. 
I'm open to revert the change if you disagree

File: src/app/Components/ArtworkRail/ArtworkRail.tsx (hunk 7 above)
Comment (dariakoko) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx:89]: Yes, this method was not returning any values which led to missing tracking data

File: src/app/Components/ArtworkRail/ArtworkRailCard.tsx (hunk 8 above)
Comment (dariakoko) [reply src/app/Components/ArtworkRail/ArtworkRailCard.tsx:88]: Nope 👍

Comment (dariakoko) [reply src/app/Components/ArtworkRail/ArtworkRailCard.tsx:88]: 😄 open to suggestions about how to make it look better..
i think, it will look better in a month or so when the experiment is over and the component will have a consistent height

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx
Comment (dariakoko) [reply src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:1]: Thanks for the catch!
Added now

File: src/app/Components/ArtworkRail/ArtworkRail.tsx (hunk 7 above)
Comment (MounirDhahri) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx:89]: thanks for explaining

File: src/app/Components/ArtworkGrids/ArtworkGridItem.tsx
Comment (MounirDhahri) [reply src/app/Components/ArtworkGrids/ArtworkGridItem.tsx:1]: thanks

File: src/app/utils/experiments/getExperimentVariant.tsx (hunk 12 above)
Comment (MounirDhahri) [reply src/app/utils/experiments/getExperimentVariant.tsx:1]: since it is currently within the `experiments` folder, I am afraid someone would use it accidentally and it would make removing it hard. Do you mind if you maybe rename it to something more specific to your feature and move it closer. Thank you

**Review by MounirDhahri — DISMISSED** [review]
thanks for addressing my comments. only one last comment left about getExperiment renaming.
